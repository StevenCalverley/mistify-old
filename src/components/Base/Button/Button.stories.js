import MyButton from "./Button.vue";

export default {
  title: "Base/Button",
  component: MyButton
  // argTypes: {
  //   backgroundColor: { control: "color" },
  //   size: { control: { type: "select", options: ["small", "medium", "large"] } }
  // }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { MyButton },
  template:
    '<my-button @onClick="onClick" v-bind="$props" >{{ label }}</my-button>'
});

export const Primary = Template.bind({});
Primary.args = {
  label: "Test"
};

// export const Secondary = Template.bind({});
// Secondary.args = {
//   label: "Button"
// };
