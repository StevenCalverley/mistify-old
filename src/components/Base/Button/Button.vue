<template>
  <button
    class="px-2 py-1 bg-blue-600 rounded text-white focus:outline-none focus:border-transparent focus:ring focus:ring-blue-400 focus:ring-opacity-75"
    @click="handleClick"
  >
    <!-- @slot Default to label prop -->
    <slot>{{ label }}</slot>
  </button>
</template>

<script>
export default {
  name: "Button",
  props: {
    /**
     * Model example
     */
    label: {
      type: String,
      default: "Label"
    }
  },
  methods: {
    handleClick() {
      /**
       * Passthrough click event
       * @type {Event}
       */
      this.$emit("onClick");
    }
  }
};
</script>

<style></style>
