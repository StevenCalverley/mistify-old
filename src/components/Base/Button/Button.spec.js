import { shallowMount, createLocalVue } from "@vue/test-utils";
import Button from "./Button.vue";

const localVue = createLocalVue();

describe("Button", () => {
  test("...renders with defaults", () => {
    const wrapper = shallowMount(Button, {
      localVue
    });
    expect(wrapper).toMatchSnapshot();
  });
});
