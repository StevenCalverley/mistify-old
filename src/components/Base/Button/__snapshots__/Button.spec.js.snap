// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button ...renders with defaults 1`] = `<button class="px-2 py-1 bg-blue-600 rounded text-white focus:outline-none focus:border-transparent focus:ring focus:ring-blue-400 focus:ring-opacity-75"></button>`;
